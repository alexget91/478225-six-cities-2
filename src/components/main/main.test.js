import React from "react";
import renderer from "react-test-renderer";
import Main from "./main";

it(`Main page correctly renders after relaunch`, () => {
  const tree = renderer
    .create(<Main offers={[{
      id: 0,
      priceByNight: 0,
      title: ``,
      type: `apartment`,
    }]}/>)
    .toJSON();

  expect(tree).toMatchSnapshot();
});
