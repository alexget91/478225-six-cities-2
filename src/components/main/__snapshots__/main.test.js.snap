// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Main page correctly renders after relaunch 1`] = `
Array [
  <h1
    className="visually-hidden"
  >
    Cities
  </h1>,
  <div
    className="tabs"
  >
    <section
      className="locations container"
    >
      <ul
        className="locations__list tabs__list"
      >
        <li
          className="locations__item"
        >
          <a
            className="locations__item-link tabs__item tabs__item--active js-city-item"
            href="#"
            onClick={[Function]}
          >
            <span>
              1
            </span>
          </a>
        </li>
        <li
          className="locations__item"
        >
          <a
            className="locations__item-link tabs__item js-city-item"
            href="#"
            onClick={[Function]}
          >
            <span>
              2
            </span>
          </a>
        </li>
      </ul>
    </section>
  </div>,
  <div
    className="cities"
  >
    <div
      className="cities__places-container container"
    >
      <section
        className="cities__places places"
      >
        <h2
          className="visually-hidden"
        >
          Places
        </h2>
        <b
          className="places__found"
        >
          1
           places to stay in 
          1
        </b>
        <form
          action="#"
          className="places__sorting"
          method="get"
        >
          <span
            className="places__sorting-caption"
          >
            Sort by
          </span>
           
          <span
            className="places__sorting-type"
            onClick={[Function]}
            tabIndex="0"
          >
            Popular
            <svg
              className="places__sorting-arrow"
              height="4"
              width="7"
            >
              <use
                xlinkHref="#icon-arrow-select"
              />
            </svg>
          </span>
          <ul
            className="places__options places__options--custom"
          >
            <li
              className="places__option places__option--active js-sorting-option"
              data-sort="popular"
              onClick={[Function]}
              tabIndex="0"
            >
              Popular
            </li>
            <li
              className="places__option js-sorting-option"
              data-sort="to-high"
              onClick={[Function]}
              tabIndex="0"
            >
              Price: low to high
            </li>
            <li
              className="places__option js-sorting-option"
              data-sort="to-low"
              onClick={[Function]}
              tabIndex="0"
            >
              Price: high to low
            </li>
            <li
              className="places__option js-sorting-option"
              data-sort="top-rated"
              onClick={[Function]}
              tabIndex="0"
            >
              Top rated first
            </li>
          </ul>
        </form>
        <div
          className="cities__places-list tabs__content places__list"
        >
          <article
            className="cities__place-card place-card"
            data-id={0}
            onMouseEnter={[Function]}
            onMouseLeave={[Function]}
          >
            
            <div
              className="cities__image-wrapper place-card__image-wrapper"
            >
              <a
                href="#"
              >
                <img
                  alt="Place image"
                  className="place-card__image"
                  height="200"
                  width="260"
                />
              </a>
            </div>
            <div
              className="place-card__info"
            >
              <div
                className="place-card__price-wrapper"
              >
                <div
                  className="place-card__price"
                >
                  <b
                    className="place-card__price-value"
                  >
                    €
                    0
                  </b>
                  <span
                    className="place-card__price-text"
                  >
                     / night
                  </span>
                </div>
                <button
                  className="place-card__bookmark-button button"
                  type="button"
                >
                  <svg
                    className="place-card__bookmark-icon"
                    height="19"
                    width="18"
                  >
                    <use
                      xlinkHref="#icon-bookmark"
                    />
                  </svg>
                  <span
                    className="visually-hidden"
                  >
                    To bookmarks
                  </span>
                </button>
              </div>
              <div
                className="place-card__rating rating"
              >
                <div
                  className="place-card__stars rating__stars"
                >
                  <span
                    style={
                      Object {
                        "width": "0%",
                      }
                    }
                  />
                  <span
                    className="visually-hidden"
                  >
                    Rating
                  </span>
                </div>
              </div>
              <h2
                className="place-card__name"
              >
                <a
                  className="js-place-name"
                  href="#"
                  onClick={[Function]}
                >
                  
                </a>
              </h2>
              <p
                className="place-card__type"
              >
                Apartment
              </p>
            </div>
          </article>
        </div>
      </section>
      <div
        className="cities__right-section"
      >
        <section
          className="cities__map map"
          id="map"
        />
      </div>
    </div>
  </div>,
]
`;
